mlflow/gateway/utils.py

```diff
@@ -126,16 +144,16 @@ def decode(cls, encoded_token: str):
         try:
             decoded_token = base64.b64decode(encoded_token)
             parsed_token = json.loads(decoded_token)
+            index = int(parsed_token.get("index"))
         except Exception as e:
             raise MlflowException.invalid_parameter_value(
                 f"Invalid SearchRoutes token: {encoded_token}"
             ) from e
 
-        index = parsed_token.get("index")
         if index is None or index < 0:
```

Comment:
```suggestion
        if index < 0:
```

We can remove `index is None` here.